src/main/java/org/mule/module/apikit/validation/body/form/transformation/MultipartFormData.java

```diff
@@ -96,11 +99,27 @@ public void addDefault(String key,String value){
     multipartEntityBuilder.addTextBody(key,value);
   }
 
-  public InputStream build() throws InvalidFormParameterException{
-    try {
-      return multipartEntityBuilder.build().getContent();
-    } catch (IOException e) {
-      throw new InvalidFormParameterException(e);
+  public MultipartFormData build(){
+    this.httpEntity = multipartEntityBuilder.build();
+    return this;
+  }
+
+  public InputStream getContent() throws InvalidFormParameterException{
+    if (httpEntity == null || httpEntity.getContentLength() < 0L) {
+      throw new InvalidFormParameterException("Content length is unknown");
+    } else {
+      try {
+        return getInputStream();
+      } catch (IOException e) {
+        throw new InvalidFormParameterException(e);
+      }
     }
   }
+
+  private InputStream getInputStream() throws IOException {
+    ByteArrayOutputStream outStream = new ByteArrayOutputStream();
```

Comment:
cant you use piped streams here?

Reply:
yes we can, I've created a POC and left a TODO to enhance the performance later.